java.net.BindException: No free port within range

Sometimes we don't have enough ports:

```
returnsErrorIfNoMatches(com.jcabi.http.mock.MkContainerTest)  Time elapsed: 2.494 sec  <<< ERROR!
java.lang.Exception: Unexpected exception, expected<java.util.NoSuchElementException> but was<java.net.BindException>
    at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.apache.maven.surefire.junitcore.pc.Scheduler$1.run(Scheduler.java:387)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.net.BindException: No free port within range: 39318=com.sun.grizzly.http.SelectorThreadHandler@4c469ca0
    at com.sun.grizzly.TCPSelectorHandler.initSelector(TCPSelectorHandler.java:432)
    at com.sun.grizzly.TCPSelectorHandler.preSelect(TCPSelectorHandler.java:378)
    at com.sun.grizzly.SelectorHandlerRunner.doSelect(SelectorHandlerRunner.java:188)
    at com.sun.grizzly.SelectorHandlerRunner.run(SelectorHandlerRunner.java:132)
    ... 3 more
```

We should not crash in that case, but wait a few seconds and try again.

--- a/src/main/java/com/jcabi/http/mock/MkGrizzlyContainer.java
+++ b/src/main/java/com/jcabi/http/mock/MkGrizzlyContainer.java
@@ -30,14 +30,11 @@
 package com.jcabi.http.mock;
 
 import com.jcabi.aspects.Loggable;
-import com.jcabi.aspects.RetryOnFailure;
 import com.jcabi.log.Logger;
 import java.io.IOException;
-import java.net.ServerSocket;
 import java.net.URI;
 import java.util.Collection;
 import lombok.EqualsAndHashCode;
-import org.glassfish.grizzly.PortRange;
 import org.glassfish.grizzly.http.server.HttpServer;
 import org.glassfish.grizzly.http.server.NetworkListener;
 import org.hamcrest.Matcher;
@@ -114,7 +111,7 @@ public final class MkGrizzlyContainer implements MkContainer {
 
     @Override
     public MkContainer start() throws IOException {
-        return this.start(MkGrizzlyContainer.reserve());
+        return this.start(0);
     }
 
     @Override
@@ -127,21 +124,20 @@ public final class MkGrizzlyContainer implements MkContainer {
                 )
             );
         }
-        this.port = prt;
         this.gws = new HttpServer();
-        this.gws.addListener(
-            new NetworkListener(
-                "grizzly",
-                NetworkListener.DEFAULT_NETWORK_HOST,
-                new PortRange(this.port)
-            )
+        final NetworkListener listener = new NetworkListener(
+            "grizzly",
+            NetworkListener.DEFAULT_NETWORK_HOST,
+            prt
         );
+        this.gws.addListener(listener);
         this.gws.getServerConfiguration().addHttpHandler(
             this.adapter,
             "/"
         );
         this.gws.start();
-        Logger.info(this, "started on port #%s", prt);
+        this.port = listener.getPort();
+        Logger.info(this, "started on port #%s", this.port);
         return this;
     }
 
@@ -166,21 +162,4 @@ public final class MkGrizzlyContainer implements MkContainer {
         this.stop();
     }
 
-    /**
-     * Reserve port.
-     * @return Reserved TCP port
-     * @throws IOException If fails
-     * @todo #126:30m Grizzly Server has native mechanism for port reservation.
-     *  See https://github.com/eclipse-ee4j/grizzly/issues/1001.
-     *  Use it and remove this method, field for port probably can be removed as
-     *  well. This should eventually close jcabi-http#116.
-     */
-    @RetryOnFailure
-    private static int reserve() throws IOException {
-        final int reserved;
-        try (ServerSocket socket = new ServerSocket(0)) {
-            reserved = socket.getLocalPort();
-        }
-        return reserved;
-    }
 }

